### DISCUSSION TITLE
podman restart container - healthcheck transient timer

### DISCUSSION THREAD

**Discussion by fornitlakon** [containers/podman#12674] [Show and tell]
On 'podman restart container' the healthcheck transient timer is not cleared resulting with an ERROR
**Failed to start transient timer unit: Unit a454fc67b6569accee4d2e65b9f9e93c34284caa5e092dce05206267a674af81.timer already exists**
i.e
`[root@localhost system]# podman restart redis
ERRO[0000] Failed to start transient timer unit: 
Unit a454fc67b6569accee4d2e65b9f9e93c34284caa5e092dce05206267a674af81.timer already exists.
`

systemd files 
/run/systemd/system/a454fc67b6569accee4d2e65b9f9e93c34284caa5e092dce05206267a674af81.service.d
/run/systemd/system/a454fc67b6569accee4d2e65b9f9e93c34284caa5e092dce05206267a674af81.timer.d

Are not cleaned on restart resulting with this ERROR

podman restart should recreate the systemd healthcheck service/timer to prevent this ERROR

Restart behavior is Ok, Container is restarted correctly 
But how can the container be restarted without this ERROR ?

**Comment by mheon** [comment]
@baude You were looking at this one earlier this week, I recall?

**Comment by baude** [comment]
You want -> https://github.com/containers/podman/pull/12697

**Reply by fornitlakon** [reply]
Look like  #12697 is a different case.
Here the healthcheck does not fail, it is successful and on container restart it is not cleaned 

Maybe this is relevant here :
[https://serverfault.com/questions/965724/make-systemd-run-fail-gracefully-if-the-unit-name-already-exists](url)

And adding **--collect** to createTimer() in
_./libpod/healthcheck_linux.go_

**Comment by fornitlakon** [comment]
@baude  @mheon Can you take a second look ?
Look like #12697 is a different case.
Here the healthcheck does not fail, it is successful and on container restart it is not cleaned

Maybe this is relevant here :
https://serverfault.com/questions/965724/make-systemd-run-fail-gracefully-if-the-unit-name-already-exists

And adding --collect to createTimer() in
./libpod/healthcheck_linux.go
